src/win/sugar.rs

```diff
@@ -66,3 +69,24 @@ impl TryFrom<Vec<u16>> for OSUnicodeString {
         })
     }
 }
+
+// RtlInitUnicodeStringEx isn't available in windows_sys at this time (see https://github.com/microsoft/win32metadata/issues/1461)
+// so we're going to roll our own. We'll rely on RtlInitUnicodeString to do this, and just make sure we don't pass it information that would
+// induce an error.
+unsafe fn init_unicode_string(
+    destination_string: *mut UNICODE_STRING,
+    source_string: *mut u16,
+) -> NTSTATUS {
+    if destination_string.is_null() && !source_string.is_null() {
+        return STATUS_INVALID_PARAMETER;
+    }
+    let mut cursor = 0;
+    while *(source_string.offset(cursor)) != 0 {
```

Comment:
Haven't checked yet, but if we have a slice where we're creating this counted string object, we should be able to check length more safely. 

Your code doesn't handle a non-0-terminated string

Reply:
Yeah the only usage of the function took a slice, so I swapped out the pointer for a slice. Simplified the definition a lot!